services/app-api/migrate.js

```diff
@@ -48,5 +48,70 @@ export const main = handler(async () => {
     })
   );
 
+  console.log("migrateTE complete");
+};
+
+const migrateRaiTimestamp = async () => {
+  // Setup query params to get all package waivers and spas
+  const params = {
+    TableName: process.env.oneMacTableName,
+    IndexName: "GSI1",
+    KeyConditionExpression: "GSI1pk IN (:pk1, :pk2)",
+    ExpressionAttributeValues: {
+      ":pk1": "OneMAC#waiver",
+      ":pk2": "OneMAC#spa",
+    },
+  };
+  const promiseItems = [];
+
+  do {
+    const results = await dynamoDb.query(params);
+    for (const item of results.Items) {
+      //get latest rai response date
+      const latestRaiResponseTimestamp = item.raiResponses?.reduce(
+        (latestRaiResponseTimestamp, currentRaiResponse) => {
+          if (
+            currentRaiResponse.submissionTimestamp > latestRaiResponseTimestamp
+          ) {
+            latestRaiResponseTimestamp = currentRaiResponse.submissionTimestamp;
+          }
+          return latestRaiResponseTimestamp;
+        }
+      );
+
+      const updateParam = {
+        TableName: process.env.oneMacTableName,
+        Key: {
+          pk: item.pk,
+          sk: item.sk,
+        },
+        UpdateExpression:
+          "SET latestRaiResponseTimestamp = :latestRaiResponseTimestamp, auditArray = list_append(:newMessage, if_not_exists(auditArray,:emptyList))",
+        ExpressionAttributeValues: {
+          ":latestRaiResponseTimestamp": latestRaiResponseTimestamp,
+          ":emptyList": [],
+          ":newMessage": [
+            `UPDATED ${Date.now()}: added latestRaiResponseTimestamp ${latestRaiResponseTimestamp}}`,
+          ],
+        },
+      };
+      promiseItems.push(updateParam);
+    }
+    params.ExclusiveStartKey = results.LastEvaluatedKey;
+  } while (params.ExclusiveStartKey);
+
+  await Promise.all(
+    promiseItems.map(async (anUpdate) => {
+      console.log("updating: ", anUpdate);
+      await dynamoDb.update(anUpdate);
+    })
+  );
+
+  console.log("migrateRaiTimestamp complete");
+};
+
+export const main = handler(async () => {
+  await migrateTE();
```

Comment:
Thank you!!!  Actually, this looks like a nice pattern to move forward with!  Great job :)